src/benchq/problem_ingestion/molecule_instance_generation.py

```diff
@@ -332,49 +437,155 @@ def _set_frozen_core_orbitals(self, molecular_data) -> mp.mp2.MP2:
+        to decide how much virtual space will be kept for the active space.
+
+        If freeze_core option is True, chemical frozen core orbitals
+        are choosen for occuped_indicies.
+
 
     Args:
-        mean_field_object: The meanfield object to be truncated.
-        ao_list: A list of atomic orbitals to use for AVAS.
-        minao: The minimum active orbital to use for AVAS.
+        geometry: A list of tuples of the form (atom, (x, y, z)) where atom is the
+            atom type and (x, y, z) are the coordinates of the atom in Angstroms.
+        basis: The basis set to use for the calculation.
+        multiplicity: The spin multiplicity of the molecule.
+        charge: The charge of the molecule.
+        avas_atomic_orbitals: A list of atomic orbitals to use for (AVAS).
+        avas_minao: The minimum active orbital to use for AVAS.
+        occupied_indices: A list of molecular orbitals not in the active space that
+            should be assumed to be fully occupied.
+        active_indices: A list of molecular orbitals to include in the active space.
+        freeze_core: A boolean specifying whether frozen core orbitals are selected
+                    for calculations.
+        fno_percentage_occupation_number: Percentage of total occupation number.
+        fno_threshold: Threshold on NO occupation numbers.
+        fno_n_virtual_natural_orbitals: Number of virtual NOs to keep.
+        scf_options: dictionary with parameters for pySCF calculations
+        mlflow_experiment_name: if supplied, pySCF calculations will be logged to
+            mlflow. See orq_workspace_id also
+        orq_workspace_id: orquestra workspace ID. Required to log mlflow info
     """
-    mean_field_object.verbose = 4
 
-    # make sure wave function is stable before we proceed
-    mean_field_object = stability(mean_field_object)
+    app_data: ChemistryApplicationData
+    scf_info: ChemistryApplicationSCFInfo
+    active_space_info: ChemistryApplicationActiveSpaceInfo
 
-    # localize before automatically selecting active space with AVAS
-    mean_field_object = localize(
-        mean_field_object, loc_type="pm"
-    )  # default is loc_type ='pm' (Pipek-Mezey)
+    def __init__(
+        self,
+        geometry: List[Tuple[str, Tuple[float, float, float]]],
+        basis: str,
+        multiplicity: int,
+        charge: int,
+        avas_atomic_orbitals: Optional[Iterable[str]] = None,
+        avas_minao: Optional[str] = None,
+        occupied_indices: Optional[Iterable[int]] = None,
+        active_indices: Optional[Iterable[int]] = None,
+        freeze_core: Optional[bool] = None,
+        fno_percentage_occupation_number: Optional[float] = None,
+        fno_threshold: Optional[float] = None,
+        fno_n_virtual_natural_orbitals: Optional[int] = None,
+        scf_options: Optional[dict] = None,
+        mlflow_experiment_name: Optional[str] = None,
+        orq_workspace_id: Optional[str] = None,
```

Comment:
**question**: if a user doesn't provide the arguments, and the `None`s are used, is it a valid use case? If yes, nvm. If not, I'd recommend to use non-optionals

Reply:
Never mind, if nothing is provided and the Nones are used, it *is* a valid use case. See `generate_hydrogen_chain_instance(2)` in the test suite. If no options are given, I'm unsure what the default behavior is.

Reply:
That's right, providing none of the optional arguments is a valid use case. This corresponds to the case where you use all of the orbitals from the SCF calculation when you generate the active space Hamiltonian.